machine/v1beta1/types_azureprovider.go

```diff
@@ -234,6 +234,10 @@ type ManagedDiskParameters struct {
 	// DiskEncryptionSet is the disk encryption set properties
 	// +optional
 	DiskEncryptionSet *DiskEncryptionSetParameters `json:"diskEncryptionSet,omitempty"`
+	// EncryptionType is the type of diisk encryption used.
+	// Possible values include: 'EncryptionAtRestWithPlatformKey', 'EncryptionAtRestWithCustomerKey', and 'EncryptionAtRestWithPlatformAndCustomerKeys'.
+	// +optional
+	EncryptionType string `json:"encryptionType,omitempty"`
```

Comment:
We tend to make enums like this a typed string with specific values as constants, the constants can then be used in implementation to validate the valid values

Reply:
As far as I know stuff in here must correspond to the Azure SDK/Client which in turn corresponds to Azure API. And there is no such filed in [the Azure API as encryption type](https://docs.microsoft.com/en-us/rest/api/compute/virtual-machines/create-or-update#manageddiskparameters).